Jenkinsfile

```diff
@@ -182,6 +197,26 @@ boolean isNative() {
     return env['NATIVE'] && env['NATIVE'].toBoolean()
 }
 
+boolean isDownstreamJob() {
+    return env['DOWNSTREAM_BUILD'] && env['DOWNSTREAM_BUILD'].toBoolean()
+}
+
+String getUpstreamTriggerProject() {
+    return env['UPSTREAM_TRIGGER_PROJECT']
+}
+
 boolean isNormalPRCheck() {
-    return !(getQuarkusBranch() || isNative())
-}
\ No newline at end of file
+    return !(isDownstreamJob() || getQuarkusBranch() || isNative())
+}
+
+boolean isUpstreamKogitoProject() {
+    return getUpstreamTriggerProject() && getUpstreamTriggerProject().startsWith('kogito')
+}
+
+boolean isUpstreamOptaplannerProject() {
+    return getUpstreamTriggerProject() && getUpstreamTriggerProject().startsWith('opta')
```

Comment:
I assume this should return `true` only if the project is 'optaplanner'? For `optawebs` or `optaplanner-quickstarts` it does not make much sense as these are the leaves in the dependency tree.

More important question, does the `getUpstreamTriggerProject()` return 'optaplanner' as the name of the repository or 'Optaplanner' in terms of the ID defined here https://github.com/kiegroup/optaplanner/pull/1304/files#diff-954fb9904df55eadd8693b9166aadf6ebb64b995f8c366ed1a1b7be597aeacb3R32 ?

Reply:
it is the repository name: https://github.com/kiegroup/kogito-pipelines/blob/master/dsl/seed/src/main/groovy/org/kie/jenkins/jobdsl/templates/KogitoJobTemplate.groovy#L336

Reply:
`I assume this should return true only if the project is 'optaplanner'? For optawebs or optaplanner-quickstarts it does not make much sense as these are the leaves in the dependency tree.`
Indeed we could just check that the Upstream repo is `optaplanner`. will update that :)